File: Python3/Medium/Accepted/zigzag-conversion_6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        
        if numRows == 1:
            return s
        n = len(s)
        res = ['' for _ in range(numRows)]
        row = 0
        flag = -1
        for i in range(n):
            res[row] += s[i]
            if row == 0 or row == numRows - 1:
                flag = -flag
            row += flag
        return ''.join(res)

File: Rust/Medium/Accepted/zigzag-conversion_6.rs
impl Solution {
    pub fn convert(s: String, num_rows: i32) -> String {
        

        let mut ans = String::new();
        let n = s.len();
        let num_rows = num_rows as usize;
        let cycle = 2 * num_rows - 2;
        if cycle == 0 {
            return s;
        }
        for i in 0..num_rows {
            let mut j = 0;
            while j + i < n {
                ans.push(s.chars().nth(j + i).unwrap());
                if i != 0 && i != num_rows - 1 && j + cycle - i < n {
                    ans.push(s.chars().nth(j + cycle - i).unwrap());
                }
                j += cycle;
            }
        }
        ans
    }
}